File: lib_ParseSceneList.js
// KEEP UP-TO-DATE WITH PYTHON VERSION
function parseSceneList(lines)
{    
    var sceneList =
    {
        //Array of obj {scene: scene, type: type}
        allScenes: new Array(),
        
        afxScenes: new Array(),
        flaScenes: new Array(),
        swfScenes: new Array(),
        manualScenes: new Array()
    };
    
    var insideBlock = true;
    //Prepass to check if START is used later and block sholdn't start from first line
    for (var i = 0; i < lines.length; i++)
    {
        var line = lines[i];
        if (line === "START")
        {
            insideBlock = false;
            break;
        } else if (line === "END")
        {
            break;
        }
    }
    
    for (var i = 0; i < lines.length; i++)
    {
        var line = lines[i];
        //Toggle should scenes be read
        if (!insideBlock && line === "START")
        {
            insideBlock = true;
            continue;
        }
        else if (insideBlock && line === "END")
        {
            insideBlock = false;
            continue;
        }
        
        if (!insideBlock) continue;
        
        //Skip comments and empty lines
        if (line.length === 0 || line.charAt(0) === "#") continue;
        
        var sceneAndType = line.split(":", 2);
        if (sceneAndType.length !== 2) throw "Scene list is malformed at line '" + i + "': expected scene:type."
        
        switch(sceneAndType[1])
        {
            case "afx":
                sceneList.afxScenes.push(sceneAndType[0]);
                break;
            case "fla":
                sceneList.flaScenes.push(sceneAndType[0]);
                break;
            case "swf":
                sceneList.swfScenes.push(sceneAndType[0]);
                break;
            case "man":
                sceneList.manualScenes.push(sceneAndType[0]);
                break;
            default:
                throw "Scene list is malformed at line '" + i + "': unknown scene type '" + sceneAndType[1] + "'";
        }
        sceneList.allScenes.push({name: sceneAndType[0], type: sceneAndType[1]});
    }
    
    return sceneList;
}

File: lib_ParseSceneList.py
# KEEP UP-TO-DATE WITH JS VERSION

import sys
import os

class SceneList:
    # List of tuples (scene, type)
    allScenes = []
    
    afxScenes = []
    flaScenes = []
    swfScenes = []
    manualScenes = []

def parseSceneList(file):
    with open(file, "r") as f:
        lines = f.read().splitlines()
    
    sceneList = SceneList()
    
    insideBlock = True
    # Prepass to check if START is used later and block sholdn't start from first line
    for line in lines:
        if line == "START":
            insideBlock = False
            break
        elif line == "END":
            break
    
    for i in range(len(lines)):
        line = lines[i]
        
        if not insideBlock and line == "START":
            insideBlock = True
            continue
        elif insideBlock and line == "END":
            insideBlock = False
            continue
        
        if not insideBlock: continue
        # Skip comments and empty lines
        if len(line) == 0 or line[0] == "#": continue
        
        sceneAndType = line.split(":", 1)
        if len(sceneAndType) != 2:
            raise ValueError("Scene list is malformed at line '" + str(i) + "': expected scene:type.")
        
        scene = sceneAndType[0]
        type = sceneAndType[1]
        
        if type == "afx":
            sceneList.afxScenes.append(scene)
        elif type == "fla":
            sceneList.flaScenes.append(scene)
        elif type == "swf":
            sceneList.swfScenes.append(scene)
        elif type == "man":
            sceneList.manualScenes.append(scene)
        else:
            raise ValueError("Scene list is malformed at line '" + str(i) + "': unknown scene type '" + type + "'")
        
        sceneList.allScenes.append((scene, type))
    
    return sceneList